code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/util/ServiceDiscoveryUtil.java

```diff
@@ -157,6 +157,12 @@ public static String getServiceSuffixFromInstanceName(String instanceName) {
             launchConfigData = DataAccessor.fields(service)
                     .withKey(ServiceConstants.FIELD_LAUNCH_CONFIG).withDefault(Collections.EMPTY_MAP)
                     .as(Map.class);
+            // if the value is empty, do not export
+            for (String key: launchConfigData.keySet()) {
```

Comment:
It is not really safe to remove key from map while iterating over it. You can introduce some temporary value for iteration (and addAll items from the original launchConfigData); and then remove form launchConfigData should work